### PR TITLE
[Lang] Support sparse solver datatype configuration

### PR DESCRIPTION
**Pull request by FantasyVR**
Related issue = #2906

### CODE REVIEW

File: taichi/program/sparse_solver.cpp [hunk 1]
```diff
@@ -55,20 +57,23 @@ bool EigenSparseSolver<EigenSolver>::info() {
...
-          MAKE_SOLVER(LLT, COLAMD),
-          MAKE_SOLVER(LDLT, AMD),
-          MAKE_SOLVER(LDLT, COLAMD),
-      };
-  std::pair<std::string, std::string> solver_key =
-      std::make_pair(solver_type, ordering);
+          MAKE_SOLVER(float32, LLT, AMD), MAKE_SOLVER(float32, LLT, COLAMD),
+          MAKE_SOLVER(float32, LDLT, AMD), MAKE_SOLVER(float32, LDLT, COLAMD)};
+  static const std::unordered_map<std::string, std::string> dt_map = {
+      {"f32", "float32"}, {"f64", "float64"}};
```
Comment (Hanke98) [inline taichi/program/sparse_solver.cpp:70]: Is float64 type supported now? Should we add float64 in `solver_factory`? (or maybe later after float64 is fully supported.

**Review by Hanke98 — COMMENTED** [review]
Sorry for the delayed response.
LGTM mostly, I only have one point that I am not so sure

BTW, should we add some test cases against data type?

**Review by strongoier — CHANGES_REQUESTED** [review]
Would you fix the compilation error on the Windows platform?

File: taichi/program/sparse_solver.cpp [hunk 2]
```diff
@@ -55,20 +57,23 @@ bool EigenSparseSolver<EigenSolver>::info() {
...
-          MAKE_SOLVER(LLT, AMD),
-          MAKE_SOLVER(LLT, COLAMD),
-          MAKE_SOLVER(LDLT, AMD),
-          MAKE_SOLVER(LDLT, COLAMD),
-      };
-  std::pair<std::string, std::string> solver_key =
-      std::make_pair(solver_type, ordering);
+          MAKE_SOLVER(float32, LLT, AMD), MAKE_SOLVER(float32, LLT, COLAMD),
+          MAKE_SOLVER(float32, LDLT, AMD), MAKE_SOLVER(float32, LDLT, COLAMD)};
+  static const std::unordered_map<std::string, std::string> dt_map = {
```
Comment (strongoier) [inline taichi/program/sparse_solver.cpp:69]: Using strings to refer to types (e.g. "f32") is not good practice, so maybe in the future some refactoring needs to be done here :-)

**Review by strongoier — APPROVED** [review]
LGTM! Only one suggestion below.

**Pull request merged**